src/SFA.DAS.EmployerIncentives.Api/Controllers/AccountCommandController.cs

```diff
@@ -37,5 +38,12 @@ public async Task<IActionResult> SignAgreement([FromRoute] long accountId, [From
             await SendCommandAsync(new SignLegalEntityAgreementCommand(accountId, accountLegalEntityId, request.AgreementVersion));
             return NoContent();
         }
+
+        [HttpPatch("/legalentities/{hashedLegalEntityId}/vendorregistrationform/status")]
```

Comment:
We shouldn't have path like this.  The path locates the resource

/legalentities/{hashedLegalEntityId}

the patched data is in the body.  So it's a REST interface rather than a RPC call.

The body should details what has changed e.g. as per https://en.wikipedia.org/wiki/JSON_Patch

So in this example would be 

[
    { "op": "replace", "path": "/vendorregistrationform/status", "value": "newStatus" }
]

Reply:
This is an interesting one, you're definitely right about it not being RESTful but what would the implementation of a generic patch method look like?  Is it worth taking this as tech debt with a spike to see how we'd implement the patch method without it becoming a bag of spanners?

Reply:
Yes a tech debt spike would be useful as the patch approach should be applied elsewhere in a consistent manor.